### ISSUE TITLE
Settings Editor is not launching in PGO tests

### ISSUE THREAD

**Issue by DHowett** [microsoft/terminal#16960]
The nightly canary build failed to PGO TerminalSettingsEditor due to no counts, which suggests that it didn't get exercised in last night's PGO run. That further hints that it may not have opened.

The PGO agents run Windows 10, I think? And they launch in unpackaged mode. May be a good place to start the investigation!

**Comment by zadjii-msft** [comment]
![image](https://github.com/microsoft/terminal/assets/18356694/f03db0db-80a5-4fd6-b014-c9cb920a5da5)


Unpackaged build fresh off of #16954 (the first PR I clicked on with a finished CI run), on a win10 vm. 

So, it's not that.

**Comment by zadjii-msft** [comment]
_notes_

```
##[warning]PGOMGR(0,0): Warning PG0188: No .PGC files matching 'C:\__w\1\s\build\PGO\..\..\packages\Microsoft.Internal.Windows.Terminal.PGODatabase.1.21.0-2403272312-main\tools\x64\Microsoft.Terminal.Settings.Editor!*.pgc' were found.
    34>PGOMGR : warning PG0188: No .PGC files matching 'C:\__w\1\s\build\PGO\..\..\packages\Microsoft.Internal.Windows.Terminal.PGODatabase.1.21.0-2403272312-main\tools\x64\Microsoft.Terminal.Settings.Editor!*.pgc' were found. [C:\__w\1\s\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Settings.Editor.vcxproj]
           Reading PGD file 1: C:\__w\1\s\build\PGO\..\..\packages\Microsoft.Internal.Windows.Terminal.PGODatabase.1.21.0-2403272312-main\tools\x64\Microsoft.Terminal.Settings.Editor.pgd
##[error]build\PGO\..\..\packages\Microsoft.Internal.Windows.Terminal.PGODatabase.1.21.0-2403272312-main\tools\x64\Microsoft.Terminal.Settings.Editor.pgd(0,0): Error C2220: the following warning is treated as an error
    34>C:\__w\1\s\build\PGO\..\..\packages\Microsoft.Internal.Windows.Terminal.PGODatabase.1.21.0-2403272312-main\tools\x64\Microsoft.Terminal.Settings.Editor.pgd : error C2220: the following warning is treated as an error [C:\__w\1\s\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Settings.Editor.vcxproj]
##[warning]build\PGO\..\..\packages\Microsoft.Internal.Windows.Terminal.PGODatabase.1.21.0-2403272312-main\tools\x64\Microsoft.Terminal.Settings.Editor.pgd(0,0): Warning C4961: No profile data was merged into 'C:\__w\1\s\build\PGO\..\..\packages\Microsoft.Internal.Windows.Terminal.PGODatabase.1.21.0-2403272312-main\tools\x64\Microsoft.Terminal.Settings.Editor.pgd', profile-guided optimizations disabled
    34>C:\__w\1\s\build\PGO\..\..\packages\Microsoft.Internal.Windows.Terminal.PGODatabase.1.21.0-2403272312-main\tools\x64\Microsoft.Terminal.Settings.Editor.pgd : warning C4961: No profile data was merged into 'C:\__w\1\s\build\PGO\..\..\packages\Microsoft.Internal.Windows.Terminal.PGODatabase.1.21.0-2403272312-main\tools\x64\Microsoft.Terminal.Settings.Editor.pgd', profile-guided optimizations disabled [C:\__w\1\s\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Settings.Editor.vcxproj]
            Creating library C:\__w\1\s\bin\x64\Release\Microsoft.Terminal.Settings.Editor\Microsoft.Terminal.Settings.Editor.lib and object C:\__w\1\s\bin\x64\Release\Microsoft.Terminal.Settings.Editor\Microsoft.Terminal.Settings.Editor.exp
         Generating code
##[error]LINK(0,0): Error LNK1257: code generation failed
    34>LINK : fatal error LNK1257: code generation failed [C:\__w\1\s\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Settings.Editor.vcxproj]

```

_abridged:_

* `PGOMGR : warning PG0188: No .PGC files matching 'C:\__w\1\s\build\PGO\..\..\packages\Microsoft.Internal.Windows.Terminal.PGODatabase.1.21.0-2403272312-main\tools\x64\Microsoft.Terminal.Settings.Editor!*.pgc'`
* `warning C4961: No profile data was merged into 'C:\__w\1\s\build\PGO\..\..\packages\Microsoft.Internal.Windows.Terminal.PGODatabase.1.21.0-2403272312-main\tools\x64\Microsoft.Terminal.Settings.Editor.pgd', profile-guided optimizations disabled `


_etc_

* LKG: [Windows Terminal OneBranch Signed Nightly_2403.26001](https://microsoft.visualstudio.com/Dart/_build/results?buildId=94336319&view=results)
  * dc64efca5e866f386d875eefbdd00ce8b9ae67bb
    * #16821 
    * #16944 
* First bad: [#Windows Terminal OneBranch Signed Nightly_2403.27001](https://microsoft.visualstudio.com/Dart/_build/results?buildId=94392215&view=results)
  * but that was dusting messing with the build
* First NIGHTLY bad: [Windows Terminal OneBranch Signed Nightly_2403.27010 ](https://microsoft.visualstudio.com/Dart/_build/results?buildId=94441713&view=results)
  * 862ff39cbadac53aa9d5c4eb3e1dd3c89ea2f8eb
    * Just the copy task 6 revert?

**Comment by DHowett** [comment]
The PGO builds trail the nightlies by one night - so the bad data would have been produced one or more nights previously in instrument pipelines

**Comment by DHowett** [comment]
The last PGO build that successfully instrumented TSE was [0.0.2403.2501](https://microsoft.visualstudio.com/Dart/_build/results?buildId=94231764&view=logs&j=70e07775-744a-56c3-524b-3cbbdadad6fe&t=657a69a2-73be-559a-29ad-82979d105c31).

The first one that failed was [0.0.2403.2601](https://microsoft.visualstudio.com/Dart/_build/results?buildId=94346419&view=logs&j=70e07775-744a-56c3-524b-3cbbdadad6fe&t=657a69a2-73be-559a-29ad-82979d105c31); evidenced by this message:

```
Running pgosweep on 'Microsoft.Terminal.Settings.Editor.dll' for test: WindowsTerminal.UIA.Tests.SmokeTests.RunOpenSettingsUI
Microsoft (R) Profile Guided Optimization Sweeping Utility 14.38.33130.0
Copyright (C) Microsoft Corporation. All rights reserved.

PGOSWEEP : fatal error PG1032: Can find no running instrumented instance of image 'Microsoft.Terminal.Settings.Editor.dll'.
```

**Comment by DHowett** [comment]
All new in -26.

```
* de7f93122 Add support for customizing font fallback (#16821)
* a67a13288 AtlasEngine: Make Direct2D/3D and Present1 configurable (#16939)
* 501522d69 Add an experimental "scratchpad pane", for testing (#16171)
* e8f18ea92 ci: trigger builds for gh-readonly-queue/* merge queue branches (#16941)
* 08dc34612 Refactor Pane to be able to host non-terminal content (#16170)
```

**Comment by zadjii-msft** [comment]
Test instrumentation builds:

* de7f93122: https://microsoft.visualstudio.com/Dart/_build/results?buildId=94842448&view=results
* 08dc34612: https://microsoft.visualstudio.com/Dart/_build/results?buildId=94842479&view=results
* main, but then just send escape at the start: https://microsoft.visualstudio.com/Dart/_build/results?buildId=94842645&view=results

**Comment by zadjii-msft** [comment]
Win10 VM, without any other terminals installed (so no cascadia code/mono anywhere)

![image](https://github.com/microsoft/terminal/assets/18356694/e41cde3f-1010-43b1-a271-32da35588565)

We suspect we broke nearby font loading, but only on windows 10.

**Comment by zadjii-msft** [comment]
This was fixed by #16993
